### PR TITLE
AddTest(cn1cn2cn3)

### PR DESCRIPTION
**Pull request by SerhiiTarasiuk**

### CODE REVIEW

File: EasyRestProjectNetTeam2/EasyRestTests/CheckNavigationTests.cs [hunk 1]
```diff
@@ -0,0 +1,92 @@
+﻿using EasyRestProjectNetTeam2.EasyRestPages;
+using NUnit.Framework;
+
+namespace EasyRestProjectNetTeam2.EasyRestTests
+{
+    class CheckNavigationTests : BaseTest
```
Comment (dmatsev) [inline EasyRestProjectNetTeam2/EasyRestTests/CheckNavigationTests.cs:6]: please specify the access modifier

File: EasyRestProjectNetTeam2/EasyRestTests/CheckNavigationTests.cs [hunk 2]
```diff
@@ -0,0 +1,92 @@
...
+            restaurantsPage = GetRestaurantsPage();
+            restaurantsPage.WaitForJonsonMenuIsClickable(dataModel.TimeToWait);
+            restaurantsPage.ClickJohnsonMenu();
+            menuPage = GetMenuPage();
+            menuPage.WaitForHotCategoryIsClickable(dataModel.TimeToWait);
+            menuPage.ClikHotCatagory();
+            var actualPageUrl = menuPage.GetPageUrl();
+            var expectedSearchWord = dataModel.NavigationHotCatagoryMenuPage;
+            StringAssert.Contains(expectedSearchWord, actualPageUrl, "Search word is missed for menu URL");
+            menuPage.HeaderMenuComponent.ClickProfileIcon();
```
Comment (dmatsev) [inline EasyRestProjectNetTeam2/EasyRestTests/CheckNavigationTests.cs]: this logic should be in the teardown method

File: EasyRestProjectNetTeam2/EasyRestTests/CheckNavigationTests.cs [hunk 3]
```diff
@@ -0,0 +1,92 @@
...
+            restaurantsPage = GetRestaurantsPage();
+            restaurantsPage.WaitForJonsonMenuIsClickable(dataModel.TimeToWait);
+            restaurantsPage.ClickJohnsonMenu();
+            menuPage = GetMenuPage();
+            menuPage.WaitForSoupCategoryIsClickable(dataModel.TimeToWait);
+            menuPage.ClikSoupCatagory();
+            var actualPageUrl = menuPage.GetPageUrl();
+            var expectedSearchWord = dataModel.NavigationSoupCatagoryMenuPage;
+            StringAssert.Contains(expectedSearchWord, actualPageUrl, "Search word is missed for menu URL");
+            menuPage.HeaderMenuComponent.ClickProfileIcon();
```
Comment (dmatsev) [inline EasyRestProjectNetTeam2/EasyRestTests/CheckNavigationTests.cs]: this logic should be in the teardown method

File: EasyRestProjectNetTeam2/EasyRestTests/CheckNavigationTests.cs [hunk 4]
```diff
@@ -0,0 +1,92 @@
...
+            restaurantsPage = GetRestaurantsPage();
+            restaurantsPage.WaitForJonsonMenuIsClickable(dataModel.TimeToWait);
+            restaurantsPage.ClickJohnsonMenu();
+            menuPage = GetMenuPage();
+            menuPage.WaitForCoctailsCategoryIsClickable(dataModel.TimeToWait);
+            menuPage.ClikCoctailsCatagory();
+            var actualPageUrl = menuPage.GetPageUrl();
+            var expectedSearchWord = dataModel.NavigationCoctailsCatagoryMenuPage;
+            StringAssert.Contains(expectedSearchWord, actualPageUrl, "Search word is missed for menu URL");
+            menuPage.HeaderMenuComponent.ClickProfileIcon();
```
Comment (dmatsev) [inline EasyRestProjectNetTeam2/EasyRestTests/CheckNavigationTests.cs]: this logic should be in the teardown method

File: EasyRestProjectNetTeam2/EasyRestTests/CheckNavigationTests.cs [hunk 5]
```diff
@@ -0,0 +1,92 @@
...
+            menuPage.HeaderMenuComponent.ClickLogOutButton();
+        }
+    }
+}
+
```
Comment (dmatsev) [inline EasyRestProjectNetTeam2/EasyRestTests/CheckNavigationTests.cs]: please remove empty lines

**Review by dmatsev — CHANGES_REQUESTED** [review]
Please resolve the comments

File: EasyRestProjectNetTeam2/EasyRestTests/CheckNavigationTests.cs (hunk 2 above)
Comment (SerhiiTarasiuk) [reply EasyRestProjectNetTeam2/EasyRestTests/CheckNavigationTests.cs]: We will add one universal method that will delete token from database in tear down. So i don't need this method.

File: EasyRestProjectNetTeam2/EasyRestTests/CheckNavigationTests.cs (hunk 3 above)
Comment (SerhiiTarasiuk) [reply EasyRestProjectNetTeam2/EasyRestTests/CheckNavigationTests.cs]: We will add one universal method that will delete token from database in tear down. So i don't need this method.

File: EasyRestProjectNetTeam2/EasyRestTests/CheckNavigationTests.cs (hunk 4 above)
Comment (SerhiiTarasiuk) [reply EasyRestProjectNetTeam2/EasyRestTests/CheckNavigationTests.cs]: We will add one universal method that will delete token from database in tear down. So i don't need this method.

File: EasyRestProjectNetTeam2/EasyRestTests/CheckNavigationTests.cs (hunk 5 above)
Comment (SerhiiTarasiuk) [reply EasyRestProjectNetTeam2/EasyRestTests/CheckNavigationTests.cs]: done

**Review by eugenek7 — APPROVED** [review]

File: EasyRestProjectNetTeam2/EasyRestTests/CheckNavigationTests.cs [hunk 6]
```diff
@@ -0,0 +1,73 @@
...
+            menuPage = GetMenuPage();
+            menuPage.WaitForSoupCategoryIsClickable(dataModel.TimeToWait);
+            menuPage.ClikSoupCatagory();
+            var actualPageUrl = menuPage.GetPageUrl();
+            var expectedSearchWord = dataModel.NavigationSoupCatagoryMenuPage;
+            StringAssert.Contains(expectedSearchWord, actualPageUrl, "Search word is missed for menu URL");
+        }
+
+        [Test]
+        public void CheckNavigationToMenuCategory()
```
Comment (eugenek7) [inline EasyRestProjectNetTeam2/EasyRestTests/CheckNavigationTests.cs]: There are 3 same names for tests

**Review by eugenek7 — CHANGES_REQUESTED** [review]
Need to change test names for CheckNavigationTests and add private setter for component on SignUpPage

**Review by eugenek7 — APPROVED** [review]

File: EasyRestProjectNetTeam2/EasyRestTests/CheckNavigationTests.cs [hunk 7]
```diff
@@ -0,0 +1,73 @@
...
+
+        [Test]
+        public void CheckNavigationToMenuCategoryHot()
+        {
+            homePage = GetHomePage();
+            homePage.HeaderMenuComponent.ClickSignInButton();
+            signInPage = GetSignInPage();
+            signInPage.SendKeysToInputEmail(dataModel.Email);
+            signInPage.SendKeysToInputPassword(dataModel.Password);
+            homePage.HeaderMenuComponent.ClickSignInButton();
```
Comment (dmatsev) [inline EasyRestProjectNetTeam2/EasyRestTests/CheckNavigationTests.cs]: You have duplicated logic - sign-in steps are present for all three tests. Please move this logic to the setup method.

Comment (SerhiiTarasiuk) [reply EasyRestProjectNetTeam2/EasyRestTests/CheckNavigationTests.cs]: Thnx. I've added one method for this.

File: EasyRestProjectNetTeam2/EasyRestPages/SignInPage.cs [hunk 8]
```diff
@@ -67,5 +68,11 @@ public void ClickCreateAccountButton()
         {
             _createAccountButton.Click();
         }
+        public void SignInWithValidData(string email, string password)
+        {
+            SendKeysToInputEmail(email);
+            SendKeysToInputPassword(password);
+            HeaderMenuComponent.ClickSignInButton();
```
Comment (dmatsev) [inline EasyRestProjectNetTeam2/EasyRestPages/SignInPage.cs]: Are you sure that this is the correct button to click? Is it located on the header menu?

File: EasyRestProjectNetTeam2/EasyRestTests/CheckNavigationTests.cs (hunk 7 above)
Comment (dmatsev) [reply EasyRestProjectNetTeam2/EasyRestTests/CheckNavigationTests.cs]: How did you resolve this? I don't see it.

File: EasyRestProjectNetTeam2/EasyRestTests/CheckSignInTests.cs [hunk 9]
```diff
@@ -16,19 +16,19 @@ public void CheckUserSignInAndLogOutTest()
...
             homePage.HeaderMenuComponent.WaitRolePanelButtonIsClickable();
             homePage.HeaderMenuComponent.ClickRolePanelButton();
             personalInfoPage = GetPersonalInfoPage();
             personalInfoPage.WaitInputEmailIsVisible();
-            Assert.AreEqual(dataModel.Email, personalInfoPage.GetTextFromEmailField(), "Emails are not equals");
+            Assert.AreEqual(dataModel.EmailForClient, personalInfoPage.GetTextFromEmailField(), "Emails are not equals");
             personalInfoPage.HeaderMenuComponent.ClickProfileIcon();
             personalInfoPage.HeaderMenuComponent.ClickLogOutButton();
-            Assert.IsTrue(signInPage.GetPageUrl().Contains(dataModel.SignInPageUrlSearchWords), "Search word is absent on SignIn page URL");
+            //Assert.IsTrue(signInPage.GetPageUrl().Contains(dataModel.SignInPageUrlSearchWords), "Search word is absent on SignIn page URL");
```
Comment (dmatsev) [inline EasyRestProjectNetTeam2/EasyRestTests/CheckSignInTests.cs]: why did you comment on this Assert?

**Review by dmatsev — COMMENTED** [review]
Please respond to my comments.

File: EasyRestProjectNetTeam2/EasyRestPages/SignInPage.cs (hunk 8 above)
Comment (SerhiiTarasiuk) [reply EasyRestProjectNetTeam2/EasyRestPages/SignInPage.cs]: This SignIn Button has the same Xpath as in Header. So I've little bit cheated. But U are right so I added this element for this page.

File: EasyRestProjectNetTeam2/EasyRestTests/CheckSignInTests.cs (hunk 9 above)
Comment (SerhiiTarasiuk) [reply EasyRestProjectNetTeam2/EasyRestTests/CheckSignInTests.cs]: This is not my test. This was misclick.

File: EasyRestProjectNetTeam2/EasyRestTests/CheckNavigationTests.cs (hunk 7 above)
Comment (SerhiiTarasiuk) [reply EasyRestProjectNetTeam2/EasyRestTests/CheckNavigationTests.cs]: Screenshot_14

File: EasyRestProjectNetTeam2/EasyRestTests/BaseTest.cs [hunk 10]
```diff
@@ -9,6 +7,8 @@
 using NUnit.Framework.Interfaces;
 using OpenQA.Selenium;
 using OpenQA.Selenium.Chrome;
+using System;
```
Comment (dmatsev) [inline EasyRestProjectNetTeam2/EasyRestTests/BaseTest.cs:10]: I believe you don't need these two namespaces.

**Review by dmatsev — APPROVED** [review]

**Pull request merged**